### ISSUE TITLE
After #28, `already closed` error

### ISSUE THREAD

**Issue by zetamatta** [josephspurrier/goversioninfo#29]
@andydotxyz and @josephspurrier :

After [We should close files, be a good citizen by andydotxyz · Pull Request #28 · josephspurrier/goversioninfo](https://github.com/josephspurrier/goversioninfo/pull/28) was merged, an error occurs like below :

```
2019/01/23 14:59:10 Error writing syso: error writing "nyagos.syso": read nyagos.ico: file already closed
```

for

```
goversioninfo.exe -file-version=4.4.0_1-7-g0275d5b -product-version=4.4.0_1-7-g0275d5b -icon=nyagos.ico -ver-major=0 -ver-minor=0 -ver-patch=0 -ver-build=0 -product-ver-major=0 -product-ver-minor=0 -product-ver-patch=0 -product-ver-build=0 -o nyagos.syso versioninfo.json
```

* versioninfo.json -> https://github.com/zetamatta/nyagos/blob/master/versioninfo.json
* nyagos.ico -> https://github.com/zetamatta/nyagos/blob/master/nyagos.ico

**Comment by josephspurrier** [comment]
I reverted the changes. Sorry, for this issue and thank you for catching it.

**Comment by josephspurrier** [comment]
@andydotxyz Can you provide info on how it caused issues for you? Once that is reproducible in a test through Travis, we'll make changes to this package. I did some searching and this package is being used by OpenShift, Microsoft, GitHub, Yubico, New Relic, and a few other large companies so I need to ensure this package is stable.

**Comment by andydotxyz** [comment]
Absolutely I understand - the reproduction steps are easy if you think of a temporary file rather than a static asset. The following steps should be executed from a single Go file:

* Create icon file
* execute the goversioninfo code
* Delete the icon

This will cause a "file in use" error (not the actual error message, apologies).
I can make a small test case if you think that would help - I am away from my Windows computer for a few hours though.

**Comment by josephspurrier** [comment]
Yes, a test case would be great. I would like to add in Windows tests using Travis now that they support Windows. There is no rush on that. I probably won't be able to do any testing until Sunday at the earliest.

**Comment by zetamatta** [comment]
> Is there any more information you can provide?

At first I found the problem when [AppVeyor's building log for my product](https://ci.appveyor.com/project/zetamatta/nyagos/builds/21821459/job/7ij6c4dax4fed0an).

I tried to see the crash on my local machine (Windows7 64bit , go version go1.11.4 windows/amd64) and found the same result (it crashes).

`copy %USERPROFILE%\go\src\github.com\josephspurrier\goversioninfo\icon.ico nyagos.ico`  and call goversioninfo.exe
  -> NG (crashed).

`cd  %USERPROFILE%\go\src\github.com\josephspurrier\goversioninfo & go test`
  -> OK (not crashed)

**Comment by andydotxyz** [comment]
I opened the new Issue #30 as on a different computer I was able to make it crash.
This may be particular to Windows io handling as I could not reproduce it on Linux...
